presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -46,15 +50,32 @@
 class QueryExecutor
         implements Closeable
 {
+    private static final JettyIoPool jettyIoPool;
```

Comment:
Create single instance in `PrestoDriver` in a `private final` field. It will be static through the `DriverManager` static reference. Than for every single `QueryExecutor` create it's own `HttpClient` with the pre-created `JettyIoPool` and inject it.